tests/v2/validation/encryption/rke2_secret_encryption_test.go

```diff
@@ -0,0 +1,168 @@
+}
+
+func (s *RKE2SecretEncryptionTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *RKE2SecretEncryptionTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.ns = "default"
+	sshCfg := &SSHConfig{}
+	config.LoadConfig(SSHConfigFileKey, sshCfg)
+	s.sshUser = sshCfg.User
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *RKE2SecretEncryptionTestSuite) TestRKE2SecretEncryptionEnabled() {
+	subSession := s.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := s.client.WithSession(subSession)
+	require.NoError(s.T(), err)
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(s.T(), clusterName, "Cluster name is not provided")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(s.T(), err)
+
+	steveClient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), steveClient)
+
+	secretTemplate := secrets.NewSecretTemplate(namegen.AppendRandomString("test-secret"), s.ns, map[string][]byte{secretKey: []byte(secretValue)}, v1.SecretTypeOpaque)
+	secretObj, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+	require.NoError(s.T(), err)
+	logrus.Infof("secret created successfully!")
+
+	envFrom := []v1.EnvFromSource{
+		{
+			SecretRef: &v1.SecretEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{Name: secretObj.Name},
+			},
+		},
+	}
+	wload := namegen.AppendRandomString("test-workload")
+	containerTemplate := workloads.NewContainer("ranchertest", "ranchertest/mytestcontainer", v1.PullAlways, []v1.VolumeMount{}, envFrom)
+	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, nil)
+	deploymentTemplate := workloads.NewDeploymentTemplate(wload, s.ns, podTemplate, true, nil)
+
+	_, err = steveClient.SteveType(workloads.DeploymentSteveType).Create(deploymentTemplate)
+	require.NoError(s.T(), err)
+
+	podResults, podErrors := pods.StatusPods(client, clusterID)
+	assert.NotEmpty(s.T(), podResults)
+	assert.Empty(s.T(), podErrors)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID)
+	require.NoError(s.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(s.T(), err)
+
+	pods, err := steveClient.SteveType(pods.PodResourceSteveType).List(nil)
+	require.NoError(s.T(), err)
+
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		"echo $testKey",
+	}
+
+	for _, pod := range pods.Data {
+		if strings.HasPrefix(pod.ObjectMeta.Name, wload) {
+			err := kwait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+				output, err := kubeconfig.KubectlExec(restConfig, pod.ObjectMeta.Name, s.ns, cmd)
+				if err != nil {
+					return false, nil
+				}
+				require.Equal(s.T(), output.String(), secretValue)
+				logrus.Info("environment variable "+secretKey+" value : ", output.String())
```

Comment:
if the last comment is true, we should add a fail case if the output doesn't match `secretValue`. Or if its encrypted here, we should check that it is via base64 package.